src/main/c/functions.c

```diff
@@ -1,44 +1,159 @@
  */
-size_t size_triangular_matrix(const size_t n) {
+size_t
+size_triangular_matrix(const size_t n) {
   return ((n - 1) * n) / 2;
 }
 
+/**
+ * Computes the distance between two particles.
+ */
+double
+compute_distance(const Particle *p1, const Particle *p2) {
+  const double x_diff = p1->x_coordinate - p2->x_coordinate;
+  const double y_diff = p1->y_coordinate - p2->y_coordinate;
+
+  return sqrt((x_diff * x_diff) + (y_diff * y_diff));
+}
+
 /**
  * Computes the overlap between two particles.
  * Note: If the overlap is negative, there is no overlap.
  */
-double compute_overlap(const Particle p1, const Particle p2) {
-  double d = p1.radious + p2.radious;
-
-  double x_diff = p1.x_coordinate - p2.x_coordinate;
-  double y_diff = p1.y_coordinate - p2.y_coordinate;
-  double distance = sqrt((x_diff * x_diff) + (y_diff * y_diff));
+double 
+compute_overlap(const Particle *p1, const Particle *p2) {
+  const double d = p1->radious + p2->radious;
+  const double distance = compute_distance(p1, p2);
 
   return d - distance;
 }
 
 /**
  * Takes two arrays of vectors, and sums together the vectors with the same index.
  */
-void sum_vectors(const size_t size, const Vector *v1, const Vector *v2, Vector *result) {
+void
+sum_vectors(const size_t size, const Vector *v1, const Vector *v2, Vector *result) {
   for (size_t i = 0; i < size; ++i) {
     result[i].x_component = v1[i].x_component + v2[i].x_component;
     result[i].y_component = v1[i].y_component + v2[i].y_component;
   }
 }
 
+/**
+ * Apply gravity to a vector of forces.
+ */
+void
+apply_gravity(const size_t size, const ParticleProperties *particles, Vector *forces) {
+  for (size_t i = 0; i < size; ++i) {
+    forces[i].y_component -= (particles[i].mass * 9.81d);
+  }
+}
+
+/**
+ * Compute the forces applied to P2 given it was collided by P1.
+ * Note: previous_normal and previous_tangent correspond to P2 with respect to P1.
+ */
+void
+collide_two_particles(const double dt, const Particle *p1, const Particle *p2, const Vector *velocity_p1, const Vector *velocity_p2, const ParticleProperties *properties_p2, Vector *force_p2, double *previous_normal, double *previous_tangent) {
+  const double distance = compute_distance(p1, p2);
+
+  const Vector normal = {
+    .x_component = (p2->x_coordinate - p1->x_coordinate) / distance,
+    .y_component = (p2->y_coordinate - p1->y_coordinate) / distance
+  };
+
+  const double normal_velocity = (normal.x_component * (velocity_p1->x_component - velocity_p2->x_component))
+      + (normal.y_component * (velocity_p1->y_component - velocity_p2->y_component));
+
+  const double tangent_velocity = (normal.y_component * (velocity_p1->x_component - velocity_p2->x_component))
+      - (normal.x_component * (velocity_p1->y_component - velocity_p2->y_component));
+
+  const double dfn = normal_velocity * properties_p2->kn * dt;
+  const double dfs = tangent_velocity * properties_p2->ks * dt;
+
+  double Fn_1_2 = *previous_normal + dfn;
```

Comment:
There are variables names like these which are not easy to read or remember. If you are physicist which is also coding this, well it is easy. But if you are reading this for the first time, it would be a waste of time trying to guess what it means. At least, these kind of variables should have comments, I think.

Reply:
Agree.
Most variables had been renamed so they are descriptive _(like `normal_velocity`)_.
Some other had not, but we now are more focused on making the code work.
I strongly believe we should give clean to the code, but maybe it would be better to do that after having something working? - Also, right now we have 3 PR in progress _(a maybe tomorrow we will start another)_. Thus, the cleaning will be better if done after all those are merged. Otherwise, we will have to do it two or three times.

> I also suggest to declare every variable in the beginning of the function.

I am against that, that will make a lot of variables that could be constants not to be it.
Given C does not support a _"only modify it on first access"_ modifier.